### ISSUE TITLE
Tye with Dapr does not inject sidecar for Worker Service projects

### ISSUE THREAD

**Issue by tonysneed** [dotnet/tye#984]
### Describe the bug
Tye with Dapr does not inject sidecar for Worker Service projects

### To Reproduce
1. Create a .NET worker service project that uses Dapr client.
2. Add a tye.yaml file that includes the Dapr exension.
3. Run `tye run` and open the Tye Dashboard at http://localhost://8000

The Tye Dashboard indicates that the worker service project does not have the Dapr sidecar.
Publishing a topic, for example, will result in an error.
```csharp
await _daprClient.PublishEventAsync("pubsub", "weather", weather);
```
Error:
```
Error starting gRPC call. HttpRequestException: No connection could be made because the target machine actively refused it. (127.0.0.1:50001)
```

Tye Dashboard:

<img width="1468" alt="tye-dapr-worker-service" src="https://user-images.githubusercontent.com/2836367/111358069-ef4cda00-8657-11eb-99da-70df9adfab74.png">

Demo project repo: https://github.com/tonysneed/Demo.DaprPubSub

### Further technical details

- The platform: Windows 10
- Output of `tye --version`: `0.6.0-alpha.21070.5+a42e4463943e3136dbd1de38474d8d62b802797c`
- Output of `tye run` command `-v debug`:

```
Loading Application Details...
Restoring and evaluating projects
Resolved metadata for service publisher at C:\Users\Tony\Source\Demo.DaprPubSub\Publisher\obj\Debug\net5.0\MicrosoftTye.ProjectMetadata.txt
Resolved metadata for service subscriber at C:\Users\Tony\Source\Demo.DaprPubSub\Subscriber\obj\Debug\net5.0\MicrosoftTye.ProjectMetadata.txt
Restore and project evaluation took: 2197.323ms
Found application version: 1.0.0
RunCommand=C:\Users\Tony\Source\Demo.DaprPubSub\Publisher\bin\Debug\net5.0\Publisher.exe
RunArguments=
TargetPath=C:\Users\Tony\Source\Demo.DaprPubSub\Publisher\bin\Debug\net5.0\Publisher.dll
PublishDir=bin\Debug\net5.0\publish\
AssemblyName=Publisher
IntermediateOutputPath=obj\Debug\net5.0\
Found target framework: net5.0
Parsed target framework name: net
Parsed target framework version: 5.0
Found shared frameworks: Microsoft.NETCore.App
IsAspNet=False
Evaluation Took: 3.9501ms
Found application version: 1.0.0
RunCommand=C:\Users\Tony\Source\Demo.DaprPubSub\Subscriber\bin\Debug\net5.0\Subscriber.exe
RunArguments=
TargetPath=C:\Users\Tony\Source\Demo.DaprPubSub\Subscriber\bin\Debug\net5.0\Subscriber.dll
PublishDir=bin\Debug\net5.0\publish\
AssemblyName=Subscriber
IntermediateOutputPath=obj\Debug\net5.0\
Found target framework: net5.0
Parsed target framework name: net
Parsed target framework version: 5.0
Found shared frameworks: Microsoft.NETCore.App, Microsoft.AspNetCore.App
IsAspNet=True
Evaluation Took: 2.1729ms
A 'placement-port' has not been defined. So the 'exclude-placement-container' will default to 'false'.
Injecting Dapr placement service...
Launching Tye Host...

[12:55:14 INF] Executing application from C:\Users\Tony\Source\Demo.DaprPubSub\tye.yaml
[12:55:14 INF] Dashboard running on http://127.0.0.1:8000
[12:55:16 INF] Docker image daprio/dapr already installed
[12:55:16 INF] Creating docker network tye_network_b023df29-8
[12:55:16 INF] Running docker command network create --driver bridge tye_network_b023df29-8
[12:55:17 INF] Running image daprio/dapr for placement_6801f0d9-1
[12:55:17 DBG] Running proxy image mcr.microsoft.com/dotnet/core/sdk:3.1 for subscriber-proxy_31ec1097-f
[12:55:17 DBG] Running proxy image mcr.microsoft.com/dotnet/core/sdk:3.1 for subscriber-dapr-proxy_b5ee2ad4-f
[12:55:17 INF] Building projects
[12:55:19 INF] Launching service publisher_9f66a75a-a: C:\Users\Tony\Source\Demo.DaprPubSub\Publisher\bin\Debug\net5.0\Publisher.exe
[12:55:19 INF] Launching service subscriber_8b99e8d3-a: C:\Users\Tony\Source\Demo.DaprPubSub\Subscriber\bin\Debug\net5.0\Subscriber.exe
[12:55:19 INF] Launching service subscriber-dapr_1fad6649-b: C:\Users\Tony/.dapr/bin/daprd.exe -app-id subscriber -app-port 59245 -dapr-grpc-port 59247 --dapr-http-port 59248 --metrics-port 59249 --placement-host-address localhost:59244
[12:55:19 INF] publisher_9f66a75a-a running on process id 18780
[12:55:19 INF] Replica publisher_9f66a75a-a is moving to a ready state
[12:55:20 INF] subscriber-dapr_1fad6649-b running on process id 19376 bound to https://localhost:59247, http://localhost:59248, http://localhost:59249
[12:55:20 INF] subscriber_8b99e8d3-a running on process id 24304 bound to http://localhost:59245, https://localhost:59246
[12:55:20 INF] Selected process 18780.
[12:55:20 INF] Listening for event pipe events for publisher_9f66a75a-a on process id 18780
[12:55:21 INF] Replica subscriber-dapr_1fad6649-b is moving to a ready state
[12:55:21 INF] Selected process 24304.
[12:55:21 INF] Listening for event pipe events for subscriber_8b99e8d3-a on process id 24304
[12:55:21 INF] Replica subscriber_8b99e8d3-a is moving to a ready state
[12:55:25 INF] Running container placement_6801f0d9-1 with ID 67e8d727b60e
[12:55:25 INF] Running docker command network connect tye_network_b023df29-8 placement_6801f0d9-1 --alias placement
[12:55:25 INF] Running container subscriber-dapr-proxy_b5ee2ad4-f with ID 2785dd5796df
[12:55:25 INF] Running docker command network connect tye_network_b023df29-8 subscriber-dapr-proxy_b5ee2ad4-f --alias subscriber-dapr
[12:55:26 INF] Running container subscriber-proxy_31ec1097-f with ID ae103d89692b
[12:55:26 INF] Running docker command network connect tye_network_b023df29-8 subscriber-proxy_31ec1097-f --alias subscriber
[12:55:26 INF] Failed to collect diagnostics for subscriber-dapr_1fad6649-b after 00:00:05.
[12:55:26 INF] Replica placement_6801f0d9-1 is moving to a ready state
[12:55:26 INF] Collecting docker logs for placement_6801f0d9-1.
[12:55:27 INF] Collecting docker logs for subscriber-dapr-proxy_b5ee2ad4-f.
[12:55:27 INF] Collecting docker logs for subscriber-proxy_31ec1097-f.
```

**Comment by jkotalik** [comment]
I think the issue is that the launchsettings.json doesn't have an http binding, so we skip assigning a binding to it. To work around this, you can either add an http binding in the tye.yaml or adding a binding to your launch settings.
```
"Publisher": {
      "commandName": "Project",
      "dotnetRunMessages": "true",
      "applicationUrl": "http://localhost:5000",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
```

For my knowledge, are you trying to only use gRPC in the worker?

**Comment by tonysneed** [comment]
Adding `"applicationUrl": "http://localhost:5000"` to launchSettings.json did not make any difference. Still no sidecar and got the same exception.

However, when I added bindings to the services in tye.yaml, the sidecar for the publisher project did appear. Then I got a different exception with `_daprClient.PublishEventAsync("pubsub", "weather", weather)`:

```
Message=Publish operation failed: the Dapr endpoint indicated a failure.
Source=Dapr.Client
Inner Exception: RpcException: Status(StatusCode="FailedPrecondition", Detail="no pubsub is configured")
```

I do not get the exception when running **dapr** directly: `dapr run --app-id publisher -- dotnet run

I can verify in the **Components** tab of the Dapr Dashboard that **pubsub** is configured properly.
